### DISCUSSION TITLE
Welcome to Prime Discussions!

### DISCUSSION THREAD

**Discussion by atkntepe** [primefaces/community#7] [Announcements]
👋 Welcome to Prime Discussions!

This is your central hub for all things related to PrimeTek's suite of open-source UI libraries, including PrimeFaces, PrimeNG, PrimeReact, and PrimeVue. Here's what you can do:

🤔 Ask questions about our products in the relevant categories.
💡 Share your ideas and suggestions.
🛠 Showcase the projects you've built with our libraries.
🇬🇧 To ensure clear communication and inclusivity, we kindly ask all members to use English when posting or commenting.
🤝 Connect with other community members and the PrimeTek team.

Let's keep this a welcoming and productive space. We can work together and make both our community and projects better.💪

**Comment by Programmercito** [comment]
Why do the buttons in the showcase have an ink effect on the button and not the latest version of PrimeNG?

**Reply by cagataycivici** [reply]
Would be better to create a separate topic for this. Thank you.

**Comment by Weicai-Ling** [comment]
Hello, 

The appearance of PrimeVUE looks exciting. I am new to PrimeVUE. Who could please advise the difference of the folders, such as pages,uikit and utilities under the views folder in sakai template? Should I create a new folder under the views folder to supply my .vue files for my application? https://github.com/primefaces/sakai-vue/tree/master/src/views

Really appreciate !

**Comment by JOROZCOF206** [comment]
Hello guys, I bought Apollo Template, so far so good. In this moment I have cleaning my template, deleting modules, deleting unnecessary things, so. I have a question, I changed the base path, that in this case is DASHBOARD, I do chaged for my base component, but when I make the change, it does not recognize the other routes, it redirects me 404. I have been checking all the modules, but I can't find how to change this base route. 

This is base path for default.

{
              path: '',
                loadChildren: () => import('./demo/components/dashboards/dashboards.module').then(m => m.DashboardsModule)
            },
But I changed for this: 
{
                path: '',
                loadChildren: () => import('./components/home/home.module').then(m => m.HomeModule)
            },
Other paths, redirect to 404. Only work it, if I add some letter o word to new path

Some help?

**Comment by sokvotha** [comment]
Hello ,  currently i have a project with PrimNG version 14 , and i want to update  my project to latest version, but now it got wrong style for verson 14 use p-grid and latest version use grid , how can i replace it to correct style ?

**Comment by bgosse2010** [comment]
I'm a little bit confused on how to get the selected values from a primevue treeselect in checkbox mode.  I see lots a examples of how to set it up but none on how to get the resulting selections.  Here's my treeslect:

`
<span class="p-float-label">
          <TreeSelect
            v-model="selectedEnterpriseProjectGroups"
            :options="nodesRef"
            selectionMode="checkbox"
            placeholder="Enterprise Project Groups"
            class="md:w-20rem w-full"
          />
          <label style="margin-left: 8px" for="dd-module"
            >Enterprise Project Groups</label
          >
        </span>
        <Button
          style="margin-top: 2px"
          @click="onEnterpriseProjectGroupsChange"
          icon="pi pi-search"
          v-tooltip.bottom="'Submit'"
        ></Button>
`

Here's the onEnterpriseProjectGroupsChange method:

`
const selectedEnterpriseProjectGroups = ref(null);
async function onEnterpriseProjectGroupsChange() {
  console.log("selectedEnterpriseProjectGroups: ", selectedEnterpriseProjectGroups.value);
  var selectedLookups = JSON.parse(JSON.stringify(selectedEnterpriseProjectGroups.value)) ;
  console.log("selectedLookups: ", selectedLookups);
}
`

My TreeNode structure seems to be ok and I can select multiple nodes from the treeselect. The selectedEnterpriseProjectGroups object looks something like this:

{
    "4": {
        "checked": false,
        "partialChecked": true
    },
    "4-0": {
        "checked": true,
        "partialChecked": false
    },
    "4-1": {
        "checked": true,
        "partialChecked": false
    },
    "4-2": {
        "checked": true,
        "partialChecked": false
    }
}

I can it has my the keys I had setup being referenced but I'm sure what I do with this. I guess I was hoping for an array that had the original data value I set for each node. Any help would be greatly appreciated.

**Reply by bgosse2010** [reply]
Well I figured out how to get the keys by using a Map:

`
const selectedEnterpriseProjectGroups = ref(null);
async function onEnterpriseProjectGroupsChange() {
  var selectedLookups = JSON.parse(
    JSON.stringify(selectedEnterpriseProjectGroups.value)
  );
  var selectedLookupsMap = new Map(Object.entries(selectedLookups));
  for (const [key, value] of selectedLookupsMap.entries()) {
    if (value.checked) {
      console.log(key, value);
    }
  }
}
`

I guess I need to do a look up in my original node tree.

**Comment by prateekjain123** [comment]
I am trying Urls which are not http or https and editor coverts them to about://blank.
Is there any way to fix it?

**Comment by ShevchukOleg** [comment]
###  optionDisabled - not working.

Hi, bug defined:
`<div class="card flex justify-content-center">
  <p-multiSelect
    [options]="cities"
    [(ngModel)]="selectedCities"
    placeholder="Select Cities"
    optionLabel="name"
    display="chip"
    [showClear]="true"
    [optionDisabled]="disabled"
  ></p-multiSelect>
</div>
<div class="card flex justify-content-center">
  <p-multiSelect
    [options]="cities2"
    [(ngModel)]="selectedCities"
    placeholder="Select Cities"
    display="chip"
    [showClear]="true"
    [optionDisabled]="disabled"
  ></p-multiSelect>
</div>`

`import { Component, OnInit } from '@angular/core';

interface City {
  name: string;
  code: string;
}

@Component({
  selector: 'multi-select-chips-demo',
  templateUrl: './multi-select-chips-demo.html',
})
export class MultiSelectChipsDemo implements OnInit {
  cities!: City[];
  cities2!: string[];
  disabled = 'New York';

  selectedCities!: City[];

  constructor() {}
  ngOnInit(): void {
    this.cities = [
      { name: 'New York', code: 'NY' },
      { name: 'Rome', code: 'RM' },
      { name: 'London', code: 'LDN' },
      { name: 'Istanbul', code: 'IST' },
      { name: 'Paris', code: 'PRS' },
    ];
    this.cities2 = ['New York', 'Rome', 'London', 'Istanbul', 'Paris'];
  }
}`

`{
  "name": "primeng-multi-select-chips-demo",
  "description": "PrimeNG MultiSelectChipsDemo",
  "license": "MIT",
  "keywords": [],
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test"
  },
  "dependencies": {
    "primeng": "17.13.0",
    "@angular-devkit/build-angular": "^17.3.1",
    "@angular/animations": "^17.3.1",
    "@angular/cdk": "^17.3.1",
    "@angular/cli": "^17.3.1",
    "@angular/common": "^17.3.1",
    "@angular/compiler": "^17.3.1",
    "@angular/compiler-cli": "^17.3.1",
    "@angular/core": "^17.3.1",
    "@angular/forms": "^17.3.1",
    "@angular/platform-browser": "^17.3.1",
    "@angular/platform-browser-dynamic": "^17.3.1",
    "@angular/platform-server": "^17.3.1",
    "@angular/router": "^17.3.1",
    "chart.js": "3.3.2",
    "primeflex": "^3.3.1",
    "primeicons": "^7.0.0",
    "quill": "1.3.7",
    "rxjs": "~7.8.1",
    "tslib": "^2.5.0",
    "zone.js": "~0.14.0"
  }
}`

**Comment by nitishmishra08** [comment]
Hi All,
I have an angular 16 application with grapesjs integration, is there any way to use PrimeNg components as custom blocks in grapesjs.

**Comment by dev123rld** [comment]
Hi,

I'm encountering an issue with Primeng charts in our project. Specifically, when I double-click on charts, it selects text from another div outside the chart surface. Do you know of any solutions for this?

Thanks!

**Comment by saulwolfdev** [comment]
Hello how are you? This week I bought the individual version of UI KIT, now it only lets me download the figma in files. It doesn't work for me like that. It's quite annoying to be downloading the new versions. I ask why they don't give me access to the ui kit library? to see it online? Or why doesn't it appear with my account to consume the components?

**Comment by BenJackGill** [comment]
You guys should add "Discussions" tab to the top of your GitHub page. Because right now the only way to find it is via Issues > Feature Request. Adding it to the top bar would attract more discussion.

Here is a screenshot for reference: 
<img width="1383" alt="Screenshot 2024-09-25 at 2 04 52 PM" src="https://github.com/user-attachments/assets/a427886c-4aab-4674-9b76-6b71520a8eca">

**Comment by carlosbh9** [comment]
Hi everyone, could someone help me to use toast notifications with primeNg in a specific version globally so that it calls the <p-toast /> command only once in the root component?

Thanks.

**Comment by exto-kamalesh** [comment]
I am reaching out regarding an issue with the p-inputNumber component in PrimeNG when used with currency formatting for certain countries, particularly Saudi Arabia and other right-to-left (RTL) languages.

Issue Details:
When selecting Saudi Arabia (SAR) as the currency, the number input behaves unexpectedly. The text cursor starts from the left instead of the right, which is incorrect for RTL languages.
The input currently only accepts numbers in the local format of the selected country and does not allow users to enter English (Western Arabic) numerals.
If an initial value is pre-filled with Arabic numerals, there is no automatic conversion to English numerals while retaining the currency symbol and format.
The correct currency symbol should always be displayed dynamically while typing, but this is not working as expected.
Expected Behavior:
The input should correctly align based on the writing direction (LTR or RTL) for the selected currency.
Users should be able to enter English numerals regardless of the selected country.
If an initial value is pre-filled with Arabic numerals, only the numbers should be converted to English while keeping the currency format intact.
Could you please advise if there is a recommended approach to handle this, or if there are any updates planned to resolve this issue?

Looking forward to your guidance.
![Recording2025-02-19134013-ezgif com-video-to-gif-converter](https://github.com/user-attachments/assets/023daae2-34a7-47e3-844b-c4094885604f)

**Comment by WizPS** [comment]
In a DataTable, how can I access the underlying data after filtering?
Specifically, when the user applies filters and I want to retrieve that filtered dataset and iterate over it in code, what is the recommended way to do this?

**Comment by Aruu-techno** [comment]
Why i am facing 500 error? The website is not starting the job of level up please help me someone
